source/NVDAObjects/UIA/__init__.py

```diff
@@ -767,19 +767,19 @@ def _get_description(self):
 			return ""
 
 	def _get_keyboardShortcut(self):
-		ret = ""
 		try:
-			ret += self.UIAElement.currentAccessKey
+			accessKey = self.UIAElement.currentAccessKey
 		except COMError:
-			pass
-		if ret:
-			#add a double space to the end of the string
-			ret +="  "
+			# #6779: Forcefully set access key to an empty string if UIA says access key is None, resolves concatenation error in focus events, object navigation and so on.
+			accessKey = ""
 		try:
-			ret += self.UIAElement.currentAcceleratorKey
+			acceleratorKey = self.UIAElement.currentAcceleratorKey
 		except COMError:
-			pass
-		return ret
+			acceleratorKey = ""
+		# #6790: Do not add two spaces unless both access key and accelerator are present in order to not waste string real estate.
+		# If not, just return the longest string.
+		keyboardShortcuts = (accessKey, acceleratorKey)
+		return "  ".join(keyboardShortcuts) if accessKey and acceleratorKey else max(keyboardShortcuts)

```

Comment:
This is clever, but I think a bit tricky to read. Since we're creating a tuple anyway, I suggest just creating a list at the start and appending the values if they aren't None/exception.